### PR TITLE
[DO NOT MERGE] feature/arm64 to main

### PR DESCRIPTION
**Pull request by WilliamXieMSFT**
!!! DO NOT MERGE !!!

###### Summary 

This feature branch brings preliminary ARM64 functionality to CLRIE. 

###### Change Log  

- Updated InstrumentationEngine core projects to build with ARM64

###### Test Methodology

local build

### CODE REVIEW

File: src/InstrumentationEngine/InstrumentationEngine.vcxproj [hunk 1]
```diff
@@ -67,7 +78,10 @@
...
-      <AdditionalDependencies>InstrumentationEngine.Lib.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;msxml6.lib;version.lib;mscoree.lib;Faultrep.lib;Shlwapi.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalLibraryDirectories Condition="'$(Platform)'=='ARM64'">$(WindowsSDK_LibraryPath_arm64);%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+      <AdditionalDependencies>InstrumentationEngine.Lib.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;msxml6.lib;version.lib;Faultrep.lib;Shlwapi.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies Condition="'$(Platform)'!='ARM64'">mscoree.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies Condition="'$(Platform)'=='ARM64'">C:\Users\willxie\Downloads\microsoft.visualstudio.arm64.sdk.17.0.0.nupkg\lib\mscoree.lib;%(AdditionalDependencies)</AdditionalDependencies>
```
Comment (wiktork) [inline src/InstrumentationEngine/InstrumentationEngine.vcxproj:84]: This needs to be changed

File: InstrumentationEngine.sln [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.28727.205
+# Visual Studio Version 17
+VisualStudioVersion = 17.2.32201.480
```
Comment (wiktork) [inline InstrumentationEngine.sln:4]: I know I mentioned this in the other PR, but what's the minimum VS version we can build this on? Ideally we can build it on the public stable dev 17 version.

File: src/InstrumentationEngine/InstrumentationEngine.vcxproj (hunk 1 above)
Comment (WilliamXieMSFT) [reply src/InstrumentationEngine/InstrumentationEngine.vcxproj:84]: We have a thread with the CLR team; they're planning to eventually release it publicly but for now we have to use an internal package.

File: InstrumentationEngine.sln (hunk 2 above)
Comment (WilliamXieMSFT) [reply InstrumentationEngine.sln:4]: We should be able to use VS2019 since our current toolset is v142 and winsdk is 10. I'll get this reverted

Comment (WilliamXieMSFT) [reply InstrumentationEngine.sln:4]: FWIW, it doesn't seem like this header does anything. I was able to open and build using the public d17.1 version using the supported toolset version.

File: src/InstrumentationEngine/InstrumentationEngine.vcxproj (hunk 1 above)
Comment (WilliamXieMSFT) [reply src/InstrumentationEngine/InstrumentationEngine.vcxproj:84]: Addressed in https://github.com/microsoft/CLRInstrumentationEngine/pull/470

**Comment by WilliamXieMSFT** [comment]
Closing this PR; comments will be addressed in separate PRs

**Pull request closed without merging**